test/net_http_test.rb

```diff
@@ -0,0 +1,218 @@
+    Semian["http_localhost_41235"].reset if Semian["http_localhost_41235"]
+    Semian.destroy('http_localhost_41234')
+    Semian.destroy('http_localhost_41235')
+    @proxy = Toxiproxy[:semian_test_net_http]
+  end
+
+  def test_server_up_and_toxiproxy_working
+    uri = URI('http://localhost:41234/200')
+    assert_equal "Success", Net::HTTP.get(uri)
+
+    uri = URI('http://localhost:41235/200')
+    @proxy.downstream(:latency, latency: 300).apply do
+      time = Time.now
+      Net::HTTP.get(uri)
+
+      time_diff = Time.now - time
+      assert time_diff >= 0.3
+    end
+  end
+
+  def test_semian_identifier
+    Net::HTTP.start("localhost", 41_235) do |http|
+      assert_equal "http_localhost_41235", http.semian_identifier
+    end
+  end
+
+  def test_trigger_open
+    open_circuit!
+    uri = URI('http://localhost:41235/200')
+    assert_raises Net::CircuitOpenError do
+      Net::HTTP.get(uri)
+    end
+  end
+
+  def test_trigger_close_after_open
+    open_circuit!
+    close_circuit!
+  end
+
+  def test_blockheads_tickets_are_working
+    threads = []
+    ticket_count = Net::HTTP.new("localhost", 41_235).raw_semian_options[:tickets]
+
+    @proxy.downstream(:latency, latency: 500).apply do
+      uri = URI('http://localhost:41235/200')
+      (ticket_count).times do
+        threads << Thread.new do
+          Net::HTTP.get(uri)
+        end
+      end
+      sleep 0.3
+      assert_raises Net::ResourceBusyError do
+        Net::HTTP.get(uri)
+      end
+
+      threads.each(&:join)
+    end
+  end
+
+  def test_get_type_1_is_protected
+    open_circuit!
+    assert_raises Net::CircuitOpenError do
+      Net::HTTP.get(URI('http://localhost:41235/200'))
+    end
+  end
+
+  def test_get_type_2_is_protected
+    open_circuit!
+    assert_raises Net::CircuitOpenError do
+      http = Net::HTTP.new("localhost", "41235")
+      http.get("/")
+    end
+  end
+
+  def test_get_type_3_is_protected
+    open_circuit!
+    assert_raises Net::CircuitOpenError do
+      uri = URI('http://localhost:41235/200')
+      Net::HTTP.get_response(uri)
+    end
+  end
+
+  def test_post_type_1_is_protected
+    open_circuit!
+    assert_raises Net::CircuitOpenError do
+      uri = URI('http://localhost:41235/200')
+      Net::HTTP.post_form(uri, 'q' => 'ruby', 'max' => '50')
+    end
+  end
+
+  def test_http_start_and_inner_methods_are_protected
+    open_circuit!
+
+    uri = URI('http://localhost:41235/200')
+    assert_raises Net::CircuitOpenError do
+      Net::HTTP.start(uri.host, uri.port) do |_|
+      end
+    end
+
+    close_circuit!
+    Net::HTTP.start(uri.host, uri.port) do |http|
+      open_circuit!
+      assert_raises Net::CircuitOpenError do
+        request = Net::HTTP::Get.new uri
+        http.request(request)
+      end
+      assert_raises Net::CircuitOpenError do
+        request = Net::HTTP::Post.new uri
+        http.request(request)
+      end
+      # and so on...
```

Comment:
There were a whole lot more, `Get`, `Post`, `Put`, `Head`, etc. I suppose I can list the ones in an array and `#each` over them.